src/app/backend/resource/deployment/deploymentdetail.go

```diff
@@ -104,8 +116,33 @@ func getDeploymentDetail(deployment *extensions.Deployment, old []*extensions.Re
 			MaxSurge:       deployment.Spec.Strategy.RollingUpdate.MaxSurge.IntValue(),
 			MaxUnavailable: deployment.Spec.Strategy.RollingUpdate.MaxUnavailable.IntValue(),
 		},
-		OldReplicaSets: oldReplicaSets,
-		NewReplicaSet:  *newReplicaSet,
-		EventList:      *events,
+		OldReplicaSetList: oldReplicaSetList,
+		NewReplicaSet:     newReplicaSet,
+		EventList:         *events,
+	}
+}
+
+func toReplicaSetList(resourceList []extensions.ReplicaSet, pods []api.Pod) replicaset.ReplicaSetList {
```

Comment:
There's a bit of duplication here on transforming extension replicaset objects into the dashboard ones. I'm not very familiar with Go or the source code, so I wasn't sure if I should, for example, export the getReplicaSetList on the replicaset package.